pkg/api/controller.go

```diff
@@ -74,2864 +61,2535 @@ type Controller struct {
+
+	response := UserList{
+		Results: make([]User, 0, len(users)),
+		Pagination: Pagination{
+			NextOffset: paginator.NextPageToken,
+			Results:    paginator.Amount,
+		},
+	}
+	for _, u := range users {
+		response.Results = append(response.Results, User{
+			Id:           u.Username,
+			CreationDate: u.CreatedAt.Unix(),
 		})
+	}
+	writeResponse(w, http.StatusOK, response)
+}
 
-		if err != nil {
-			return repositories.NewListRepositoriesUnauthorized().WithPayload(responseErrorFrom(err))
-		}
-		c.LogAction(ctx, "list_repos")
+func (c *Controller) DeleteGroupMembership(w http.ResponseWriter, r *http.Request, groupID string, userID string) {
+	if !c.authorize(w, r, []permissions.Permission{
+		{
+			Action:   permissions.RemoveGroupMemberAction,
+			Resource: permissions.GroupArn(groupID),
+		},
+	}) {
+		return
+	}
 
-		after, amount := getPaginationParams(params.After, params.Amount)
+	ctx := r.Context()
+	c.LogAction(ctx, "remove_user_from_group")
+	err := c.Auth.RemoveUserFromGroup(ctx, userID, groupID)
+	if handleAPIError(w, err) {
+		return
+	}
+	writeResponse(w, http.StatusNoContent, nil)
+}
 
-		repos, hasMore, err := c.Catalog.ListRepositories(ctx, amount, after)
-		if err != nil {
-			return repositories.NewListRepositoriesDefault(http.StatusInternalServerError).
-				WithPayload(responseError("error listing repositories: %s", err))
-		}
+func (c *Controller) AddGroupMembership(w http.ResponseWriter, r *http.Request, groupID string, userID string) {
+	if !c.authorize(w, r, []permissions.Permission{
+		{
+			Action:   permissions.AddGroupMemberAction,
+			Resource: permissions.GroupArn(groupID),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "add_user_to_group")
+	err := c.Auth.AddUserToGroup(ctx, userID, groupID)
+	if handleAPIError(w, err) {
+		return
+	}
+	writeResponse(w, http.StatusCreated, nil)
+}
 
-		repoList := make([]*models.Repository, len(repos))
-		var lastID string
-		for i, repo := range repos {
-			repoList[i] = &models.Repository{
-				StorageNamespace: repo.StorageNamespace,
-				CreationDate:     repo.CreationDate.Unix(),
-				DefaultBranch:    repo.DefaultBranch,
-				ID:               repo.Name,
-			}
-			lastID = repo.Name
-		}
-		returnValue := repositories.NewListRepositoriesOK().WithPayload(&repositories.ListRepositoriesOKBody{
-			Pagination: &models.Pagination{
-				HasMore:    swag.Bool(hasMore),
-				Results:    swag.Int64(int64(len(repoList))),
-				MaxPerPage: swag.Int64(MaxResultsPerPage),
-			},
-			Results: repoList,
-		})
-		if hasMore {
-			returnValue.Payload.Pagination.NextOffset = lastID
-		}
+func (c *Controller) ListGroupPolicies(w http.ResponseWriter, r *http.Request, groupID string, params ListGroupPoliciesParams) {
+	if !c.authorize(w, r, []permissions.Permission{
+		{
+			Action:   permissions.ReadGroupAction,
+			Resource: permissions.GroupArn(groupID),
+		},
+	}) {
+		return
+	}
 
-		return returnValue
+	ctx := r.Context()
+	c.LogAction(ctx, "list_user_policies")
```

Comment:
For example, an action name error here.

Reply:
agree - if it was aligned with the request OperationID it would have done automatically.
Can be aligned after adding some legacy to the BI.